examples/02_07_specific_words_with_tfidf.ipynb

```diff
@@ -0,0 +1,198 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "id": "E7oBUANqIGzm"
+   },
+   "outputs": [],
+   "source": [
+    "from textsemantics.server_api import ServerAPI\n",
+    "import string\n",
+    "import nltk\n",
+    "from nltk.tokenize import RegexpTokenizer\n",
+    "from nltk.corpus import stopwords\n",
+    "from lemmagen.lemmatizer import Lemmatizer\n",
+    "from lemmagen import DICTIONARY_SLOVENE\n",
+    "from flair.data import Sentence\n",
+    "from flair.embeddings import WordEmbeddings, DocumentPoolEmbeddings\n",
+    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
+    "import numpy as np\n",
+    "\n",
+    "nltk.download('stopwords', quiet=True)\n",
+    "\n",
+    "def preprocess(corpus):\n",
+    "    stop_words = set(stopwords.words('slovene'))\n",
+    "    tokenizer = RegexpTokenizer(\"\\w+\")\n",
+    "    lemmatizer = Lemmatizer(dictionary=DICTIONARY_SLOVENE)\n",
+    "    \n",
+    "    preprocessed = list()\n",
+    "    for text in corpus:\n",
+    "        text = text.translate(text.maketrans('', '', string.punctuation))\n",
+    "        tokens = tokenizer.tokenize(text.lower())\n",
+    "        tokens = [lemmatizer.lemmatize(token) for token in tokens if token not in stop_words \n",
+    "                  and len(token) > 2 and not token.isnumeric()]\n",
+    "        preprocessed.append(tokens)\n",
+    "        \n",
+    "    return preprocessed\n",
+    "\n",
+    "api = ServerAPI()\n",
+    "datasets = api.list_datasets()\n",
+    "metadata = api.get_metadata(datasets[2][0], sample_size=100, sampling_strategy='latest')\n",
+    "\n",
+    "texts = api.get_texts(urls=metadata['text'])\n",
+    "texts = [text for text in texts if len(text) > 50]\n",
+    "\n",
+    "tokens_list = preprocess(texts)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "id": "dLcquNeRIl0a"
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
+    "\n",
+    "joined_texts = [' '.join(tokens) for tokens in tokens_list]\n",
+    "vectorizer = TfidfVectorizer()\n",
+    "X = vectorizer.fit_transform(joined_texts)\n",
+    "feature_names = vectorizer.get_feature_names()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_tfidf_words(document_id, tfidf, feature_names, n_words=5):\n",
+    "    n_words = n_words if n_words and n_words <= len(feature_names) else len(feature_names)\n",
+    "    feature_index = tfidf[document_id,:].nonzero()[1]\n",
+    "    tfidf_scores = zip(feature_index, [tfidf[document_id, x] for x in feature_index])\n",
+    "    return sorted([(feature_names[i], s) for (i, s) in tfidf_scores], key=lambda tup: tup[1], reverse=True)[:n_words]"
```

Comment:
These two lines can be a bit simplified to avoid zip and one list comprehension:
```
    tfidf_scores = [(feature_names[x], tfidf[document_id, x]) for x in feature_index]
    return sorted(tfidf_scores, key=lambda tup: tup[1], reverse=True)[:n_words]
```

Reply:
Ok, will make it nicer. :)